### PR TITLE
Using regex to clean the title of the posts

### PR DESCRIPTION
**Pull request by MohammadOsama15**
This is how the data looked before, with all the signs 
image

cleans the data and looks like this, this is just an example but main use it to preprocess data 
image

### CODE REVIEW

File: redditAPI.py [hunk 1]
```diff
@@ -36,6 +38,15 @@ def get_access_token():
         print(f"Error: {e}")
         return None
 
+
+def clean_title(title):
+    # Remove non-ASCII characters
```
Comment (MahreenAsghar) [inline redditAPI.py]: remove all unicode characters instead of just non-ASCII characters.

File: redditAPI.py [hunk 2]
```diff
@@ -74,6 +85,7 @@ def get_posts(query, cap=None):
         for post in posts:
             title = post['data'].get('title')
             if title is not None and title not in titles:
+                title = clean_title(title)
```
Comment (MahreenAsghar) [inline redditAPI.py]: Don't send the cleaned titles to display, save them in a different array that can be used to do ML. Display original titles on the search page.

File: redditAPI.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 import os
 from dotenv import load_dotenv
 import time
+import re
```
Comment (MahreenAsghar) [inline redditAPI.py]: Please delete this import if not using this library.

File: redditAPI.py (hunk 2 above)
Comment (MohammadOsama15) [reply redditAPI.py]: Done!

File: redditAPI.py (hunk 1 above)
Comment (MohammadOsama15) [reply redditAPI.py]: Done, taken care of!

File: redditAPI.py (hunk 3 above)
Comment (MohammadOsama15) [reply redditAPI.py]: Forgot about it, Thanks will do and recommit!

**Review by MahreenAsghar — COMMENTED** [review]
This looks good!

**Review by MahreenAsghar — APPROVED** [review]
This looks good!

**Pull request merged**